vunk-lexer/src/lib.rs

```diff
@@ -98,40 +98,41 @@ pub fn lexer<'src>() -> impl Parser<'src, &'src str, Vec<Spanned<Token<'src>>>,
         // Put the "long" tokens first, so they get parsed first
         chumsky::primitive::choice((
             chumsky::primitive::choice((
-                just("->").map(Token::Op),
-                just("==").map(Token::Op),
-                just("!=").map(Token::Op),
-                just("<=").map(Token::Op),
-                just(">=").map(Token::Op),
-                just("&&").map(Token::Op),
-                just("||").map(Token::Op),
-                just("++").map(Token::Op),
+                just("->"),
+                just("=="),
+                just("!="),
+                just("<="),
+                just(">="),
+                just("&&"),
+                just("||"),
+                just("++"),
             )),
             chumsky::primitive::choice((
-                just("<").map(Token::Op),
-                just(">").map(Token::Op),
-                just("&").map(Token::Op),
-                just("|").map(Token::Op),
-                just("(").map(Token::Op),
-                just(")").map(Token::Op),
-                just(",").map(Token::Op),
-                just("=").map(Token::Op),
-                just(":").map(Token::Op),
-                just(".").map(Token::Op),
-                just(";").map(Token::Op),
-                just("[").map(Token::Op),
-                just("]").map(Token::Op),
-                just("{").map(Token::Op),
-                just("}").map(Token::Op),
-                just("|").map(Token::Op),
-                just("+").map(Token::Op),
-                just("-").then_ignore(just(">").not()).map(Token::Op),
-                just("*").map(Token::Op),
-                just("/").map(Token::Op),
-                just("%").map(Token::Op),
-                just("^").map(Token::Op),
+                just("<"),
+                just(">"),
+                just("&"),
+                just("|"),
+                just("("),
+                just(")"),
+                just(","),
+                just("="),
+                just(":"),
+                just("."),
+                just(";"),
+                just("["),
+                just("]"),
+                just("{"),
+                just("}"),
+                just("|"),
+                just("+"),
+                just("-").then_ignore(just(">").not()),
+                just("*"),
+                just("/"),
+                just("%"),
+                just("^"),
```

Comment:
Since each of these cases only has a single character, you can also do

```rs
one_of("<>&|(),=:.;[]{}|+*/%^")
    .or(just("-").then_ignore(just(">").not()))
    map(Token::Op)
```

if you wanted to. No pressure though, I'm sure `rustc` will probably flatten them down to something very similar, if not identical.